lib/src/core/room.dart

```diff
@@ -154,6 +124,34 @@ class Room extends DisposableChangeNotifier with EventsEmittable<RoomEvent> {
       events.emit(const RoomReconnectingEvent());
     })
     ..on<EngineDisconnectedEvent>((event) => _handleClose())
+    ..on<SignalConnectionStateUpdatedEvent>((event) {
+      // during reconnection, need to send sync state upon signal connection.
+      if (event.didReconnect) {
+        logger.fine('Sending syncState');
+        _sendSyncState();
+      }
+    })
+    ..on<SignalJoinResponseEvent>((event) {
+      sid = event.response.room.sid;
```

Comment:
maybe introduce a local variable like `joinResponse = event.response` and use that. Will make it easier to read code :-)